web/client/actions/catalog.js

```diff
@@ -256,12 +260,15 @@ export function describeError(layer, error) {
     };
 }
 
-export function addLayerAndDescribe(layer) {
+export function addLayerAndDescribe(layer, {zoomToLayer = false} = {}) {
     return (dispatch, getState) => {
         const state = getState();
         const layers = layersSelector(state);
         const id = LayersUtils.getLayerId(layer, layers || []);
         dispatch(addNewLayer({...layer, id}));
+        if (zoomToLayer && layer.bbox) {
```

Comment:
The zoom to layer was triggered by UI tool using the information from catalog record.
TMS catalog doesn't have the bbox in the catalog record (only doing a second request, that is fetched when add is dispatched). 
Moving it from UI to side effect system is also more clear (because it is a side effect)

Reply:
ok, I didn't see you removed it from UI, and just moved it here, then, right?

Reply:
Sure.
Here: 
https://github.com/geosolutions-it/MapStore2/pull/5089/files/edc819bfb71d7a6317146e657c69dd451f99dd85#diff-bfde3c9a1a2b489e99045fcc1fb257ccL366